nomad/deploymentwatcher/deployments_watcher_test.go

```diff
@@ -33,6 +33,10 @@ func TestWatcher_WatchDeployments(t *testing.T) {
 	require := require.New(t)
 	w, m := defaultTestDeploymentWatcher(t)
 
+	m.On("UpdateDeploymentStatus", mocker.MatchedBy(func(args *structs.DeploymentStatusUpdateRequest) bool {
+		return true
+	})).Return(nil).Maybe()
```

Comment:
I considered adding this boilerplate in the "constructor", `testDeploymentWatcher`, but opted against it.   Some tests have custom assertions on `UpdateDeploymentStatus`, and testify mocking docs[1] don't spec how it handles multiple `On` calls for the same method name.  Empirically, I found that the error messages get super confusing.

Instead, I opted to ensure that every test gets one `m.On("UpdateDeploymentStatus"...)` mocking setup call.

[1] https://pkg.go.dev/github.com/stretchr/testify/mock?tab=doc

Reply:
That makes sense!